### ISSUE TITLE
fails to create new folder/file with drupal.org patch

### ISSUE THREAD

**Issue by tobybellwood** [mageops/php-composer-plugin-patchset#1]
Given the patch https://www.drupal.org/files/issues/2018-06-26/events_log_track-add-workflows-2974170-7.patch (which adds a new file into a new folder) - the patch fails to apply.

Is this not a valid patch for composer-plugin-patchset, given that other (non-new-file) patches apply ok - is there anything that I should be trying to diagnose (the same patch applies fine via composer-patches?

```
Executing command (CWD): patch '--posix' '--batch' '--forward' '--strip=1' '--input=/Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/govcms/patchset/drupal/event_log_track/increase-character-length-2930817-2.patch' '--directory=web/modules/contrib/events_log_track' '--remove-empty-files'
Applied patch govcms/patchset:drupal/event_log_track/increase-character-length-2930817-2.patch [*] (https://www.drupal.org/files/issues/increase-character-length-2930817-2.patch) using patch method

Executing command (CWD): patch '--posix' '--batch' '--forward' '--strip=1' '--input=/Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/govcms/patchset/drupal/event_log_track/events_log_track-add-workflows-2974170-7.patch' '--directory=web/modules/contrib/events_log_track' '--remove-empty-files'
Failed to apply patch govcms/patchset:drupal/event_log_track/events_log_track-add-workflows-2974170-7.patch [*] (https://www.drupal.org/files/issues/2018-06-26/events_log_track-add-workflows-2974170-7.patch) using patch method

                                                                                                                                                                                                                                             
  [Creativestyle\Composer\Patchset\Exception\PatchApplicationFailedException]                                                                                                                                                                
  Could not apply patch - command "patch '--posix' '--batch' '--forward' '--strip=1' '--input=/Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/govcms/patchset/drupal/event_log_track/events_log_track-add-workflows-2974170-7.patch'   
  '--directory=web/modules/contrib/events_log_track' '--remove-empty-files'" failed with:                                                                                                                                                    
  can't find file to patch at input line 6                                                                                                                                                                                                   
  Perhaps you used the wrong -p or --strip option?                                                                                                                                                                                           
  The text leading up to this was:                                                                                                                                                                                                           
  --------------------------                                                                                                                                                                                                                 
  |diff --git a/event_log_track_workflows/event_log_track_workflows.info.yml b/event_log_track_workflows/event_log_track_workflows.info.yml                                                                                                  
  |new file mode 100644                                                                                                                                                                                                                      
  |index 0000000..5ead5ed                                                                                                                                                                                                                    
  |--- /dev/null                                                                                                                                                                                                                             
  |+++ b/event_log_track_workflows/event_log_track_workflows.info.yml                                                                                                                                                                        
  --------------------------                                                                                                                                                                                                                 
  No file to patch.  Skipping patch.                                                                                                                                                                                                         
  1 out of 1 hunk ignored                                                                                                                                                                                                                    
  can't find file to patch at input line 26                                                                                                                                                                                                  
  Perhaps you used the wrong -p or --strip option?                                                                                                                                                                                           
  The text leading up to this was:                                                                                                                                                                                                           
  --------------------------                                                                                                                                                                                                                 
  |diff --git a/event_log_track_workflows/event_log_track_workflows.module b/event_log_track_workflows/event_log_track_workflows.module                                                                                                      
  |new file mode 100755                                                                                                                                                                                                                      
  |index 0000000..0f4038a                                                                                                                                                                                                                    
  |--- /dev/null                                                                                                                                                                                                                             
  |+++ b/event_log_track_workflows/event_log_track_workflows.module                                                                                                                                                                          
  --------------------------                                                                                                                                                                                                                 
  No file to patch.  Skipping patch.                                                                                                                                                                                                         
  1 out of 1 hunk ignored                                                                                                                                                                                                                    
                                                                                                                                                                                                                                             

Exception trace:
 () at /Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/creativestyle/composer-plugin-patchset/src/PatchApplicator.php:176
 Creativestyle\Composer\Patchset\PatchApplicator->applyPatch() at /Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/creativestyle/composer-plugin-patchset/src/Patcher.php:309
 Creativestyle\Composer\Patchset\Patcher->applyPatches() at /Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/creativestyle/composer-plugin-patchset/src/Patcher.php:388
 Creativestyle\Composer\Patchset\Patcher->patch() at /Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/creativestyle/composer-plugin-patchset/src/Plugin.php:71
 Creativestyle\Composer\Patchset\Plugin->applyPatches() at /Users/toby/Sites/lagoon/govcms8-scaffold-paas/vendor/creativestyle/composer-plugin-patchset/src/Plugin.php:55
 Creativestyle\Composer\Patchset\Plugin->onPreAutoloadDump() at n/a:n/a
 call_user_func() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:176
 Composer\EventDispatcher\EventDispatcher->doDispatch() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:96
 Composer\EventDispatcher\EventDispatcher->dispatchScript() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/src/Composer/Autoload/AutoloadGenerator.php:112
 Composer\Autoload\AutoloadGenerator->dump() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/src/Composer/Installer.php:304
 Composer\Installer->run() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/src/Composer/Command/InstallCommand.php:122
 Composer\Command\InstallCommand->execute() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/vendor/symfony/console/Command/Command.php:245
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/vendor/symfony/console/Application.php:835
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/vendor/symfony/console/Application.php:185
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/src/Composer/Console/Application.php:267
 Composer\Console\Application->doRun() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/src/Composer/Console/Application.php:106
 Composer\Console\Application->run() at phar:///usr/local/Cellar/composer/1.9.3/bin/composer/bin/composer:61
 require() at /usr/local/Cellar/composer/1.9.3/bin/composer:24
```

**Comment by pinkeen** [comment]
Hello, thanks for the report!
Yes, I think this is a bug. We've never actually encountered this kind of use case (adding a new file).
I will look into this shortly.

**Comment by pinkeen** [comment]
I have found the issue, I pass `--posix` switch to the `patch` command - I thought it'd be good idea to  enable it to have the patching process behave more strictly. It seems that strict POSIX patches do not support creation of new files. It's working fine without this switch.

I am still investigating, however, I think I will just remove this switch as the benefits of having it are purely theoretical and outweighed by the incompatibilities it introduces.

**Comment by pinkeen** [comment]
Fixed in `v2.1.0` 🎉

**Comment by pinkeen** [comment]
Fixed in version tagged `v2.1.0` by commit 44fc624b8208cb12188f719a9830fc3fa23b1496

**Comment by tobybellwood** [comment]
Beautiful - that works perfectly - thanks @pinkeen!
